test/test_devices.py

```diff
@@ -40,6 +43,56 @@ def test_sync(self):
 
     self.assertEqual(met.counter_value('MarkStep'), 1)
 
+  def test_step(self):
+    with xla.step():
+      torch.ones((3, 3), device=xla.device())
+
+    self.assertEqual(met.counter_value('MarkStep'), 1)
+
+  def test_step_exception(self):
+    with self.assertRaisesRegex(RuntimeError, 'Expected error'):
+      with xla.step():
+        torch.ones((3, 3), device=xla.device())
+        raise RuntimeError('Expected error')
+
+    self.assertEqual(met.counter_value('MarkStep'), 1)
+
+  # Should roughly match example given in README
+  def test_trivial_model(self):
+
+    class TrivialModel(nn.Module):
+
+      def __init__(self):
+        super().__init__()
+        self.linear = nn.Linear(10, 10)
+
+      def forward(self, x):
+        return self.linear(x)
+
+    model = TrivialModel().to(xla.device())
+
+    batch_size = 16
+    num_samples = 100
+
+    input_data = torch.randn(num_samples, 10)
+    target_data = torch.randn(num_samples, 10)
+
+    # Create a DataLoader
+    dataset = TensorDataset(input_data, target_data)
+    loader = DataLoader(dataset, batch_size=batch_size, shuffle=True)
```

Comment:
does DataLoader don't take device as argument?

Reply:
No. In normal PyTorch, you have to move the data with `tensor.to`: https://pytorch.org/tutorials/beginner/blitz/cifar10_tutorial.html#training-on-gpu